front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -255,15 +257,24 @@ export default function AssistantBuilder({
           modelConfig.contextSize * 0.9
         }`
       );
-      setInstructionsError(`Instructions may exceed context size window.`);
-      valid = false;
-    } else {
-      setInstructionsError(null);
+      localInstructionError = `Instructions may exceed context size window.`;
     }
 
-    if (!builderState.actions.every((a) => hasActionError(a) === null)) {
-      valid = false;
-    }
+    // We only keep the last error. If there are multiple errors, the user will have to fix them one by one.
```

Comment:
NIT : If we keep only one error we could do
```js
if (error 1) {
   localError = "1";
} else if (error2) {
  localError = "2";
} else if (error3) {
  localError = "3";
}

instead of computing them all even if error 1.

Reply:
Yep, you're right, fixing !